cluster-autoscaler/cloudprovider/digitalocean/digitalocean_node_group.go

```diff
@@ -113,10 +113,10 @@ func (n *NodeGroup) DeleteNodes(nodes []*apiv1.Node) error {
 	for _, node := range nodes {
 		nodeID, ok := node.Labels[nodeIDLabel]
 		if !ok {
-			// CA creates fake node objects to represent upcoming VMs that haven't
-			// registered as nodes yet. They have node.Spec.ProviderID set. Use
-			// that as nodeID.
-			nodeID = node.Spec.ProviderID
+			// CA creates fake node objects to represent upcoming VMs that
+			// haven't registered as nodes yet. We cannot delete the node at
+			// this point.
+			return fmt.Errorf("cannot delete node %q on node pool %q: node ID label %q is missing", node.Name, n.id, nodeIDLabel)
```

Comment:
Should we include the provider id in this error message? Was just thinking of one of the cases where an orphaned droplet may attached to the master without being properly tracked by the DOKS DB.

Reply:
Yep good call -- it's now part of the message.